pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/ASTExpression.java

```diff
@@ -20,4 +20,30 @@ public ASTExpression(JavaParser p, int id) {
     public Object jjtAccept(JavaParserVisitor visitor, Object data) {
         return visitor.visit(this, data);
     }
+
+    public boolean isStandAlonePrimitive() {
```

Comment:
I'm a little confused by this method.... It is named as to recognized primitives, but will actually recognize literals, that are not necessarily primitives (ie: a String literal). Also, the last check `literal.jjtGetNumChildren() != 0` means we are not considering the primitive booleans (`true` and `false`) as stand alone primitives.

`null` would also not be recognized as a primitive, but that one is probably in line with the method name.

I'm not sure how much of the name should be changed vs the implementation, but definitely needs some rework